src/node_url.cc

```diff
@@ -245,16 +248,22 @@ void BindingData::Parse(const FunctionCallbackInfo<Value>& args) {
   if (args[1]->IsString()) {
     base_ = Utf8Value(isolate, args[1]).ToString();
     base = ada::parse<ada::url_aggregator>(*base_);
-    if (!base) {
+    if (!base && raise_exception) {
       return ThrowInvalidURL(realm->env(), input.ToStringView(), base_);
+    } else if (!base) {
+      return args.GetReturnValue().SetNull();
     }
     base_pointer = &base.value();
   }
   auto out =
       ada::parse<ada::url_aggregator>(input.ToStringView(), base_pointer);
 
   if (!out) {
-    return ThrowInvalidURL(realm->env(), input.ToStringView(), base_);
+    if (raise_exception) {
+      return ThrowInvalidURL(realm->env(), input.ToStringView(), base_);
+    } else {
+      return args.GetReturnValue().SetNull();
```

Comment:
I'm not sure but returning undefined might be cheaper

```suggestion
      return;
```

Reply:
resolved